### ISSUE TITLE
Cannot open external links.

### ISSUE THREAD

**Issue by dbrennand** [microsoft/vscode#84265]
- VSCode Version: 1.40.0
- OS Version: Windows_NT x64 10.0.018362

Steps to Reproduce:

1. Open a blank editor.
2. Type a website eg: https://google.com
3. CTRL+Click to open prompt:
![image](https://user-images.githubusercontent.com/52419383/68479853-40aab500-022b-11ea-9cf4-6c3c0502f751.png)
4. Click **Open Link**
5. Nothing happens 😢 

Does this issue occur when all extensions are disabled?: Yes

Checked with Insiders build. Same issue.

**Comment by gjsjohnmurray** [comment]
/confirmationPending

I wasn't able to reproduce this. The link opened in my default browser, Firefox.

My version info:
```
Version: 1.40.0 (user setup)
Commit: 86405ea23e3937316009fc27c9361deee66ffbf5
Date: 2019-11-06T17:02:13.381Z
Electron: 6.1.2
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Windows_NT x64 10.0.17134
```

**Comment by octref** [comment]
@dbrennand If you do this, can you open the link then?

- Run "Manage Trusted Domains"
- Add `https://google.com`
- Open the link again

**Comment by dbrennand** [comment]
My full version info:
```
Version: 1.40.0 (system setup)
Commit: 86405ea23e3937316009fc27c9361deee66ffbf5
Date: 2019-11-06T17:02:13.381Z
Electron: 6.1.2
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Windows_NT x64 10.0.18362```

Added `https://google.com` to my trusted domains as suggest by @octref, `CTRL+Click` no longer produces any prompt and browser still doesn't open the page.

**Comment by asim-rehman** [comment]
I also wasn't able to reproduce this issue. 

**Steps i took to reproduce issue:**

1. Opened blank editor
2. Added Google URL into blank editor
3. Clicked It (CTRL+Mouse Click)
4. Clicked Open Link
5. Opened successfully in Google Chrome (Default Browser)

**System Information**
Version: 1.40.0 (system setup)
Commit: 86405ea23e3937316009fc27c9361deee66ffbf5
Date: 2019-11-06T17:02:13.381Z
Electron: 6.1.2
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Windows_NT x64 10.0.18362

**Comment by octref** [comment]
@dbrennand Then that's not related to the link protection. Can you open Help -> Developer Tools to see if there's any output there when you click the link?

**Comment by dbrennand** [comment]
@octref The following errors occur when I click the link.

![image](https://user-images.githubusercontent.com/52419383/68759639-f6994900-0607-11ea-8511-4e7f7550f6ee.png)

**Comment by octref** [comment]
@dbrennand Does it help if you try to change default browser to something else?

Also, in developer tools, if you execute

```js
require('electron').shell.openExternal('https://google.com')
```

Does it produce the same error?

**Comment by dbrennand** [comment]
@octref Changing the default browser didn't not work 😞 
Furthermore running `require('electron').shell.openExternal('https://google.com')` produced the following error:
![image](https://user-images.githubusercontent.com/52419383/68850402-e21e8480-06cb-11ea-8be8-ee3dcf47b7a3.png)

**Comment by octref** [comment]
Seems to be an electron issue on your system. Not sure what I can do with it. /cc @deepak1556

**Comment by deepak1556** [comment]
@dbrennand its most likely a default browser is not set on your device, what does running the following executable print on the command line ?
[open-external.zip](https://github.com/microsoft/vscode/files/3849958/open-external.zip)

Its the following exact equivalent of what electron does behind the scenes for above api.

```
#include <iostream>
#include "windows.h"
#include "shellapi.h"

int main() {
  auto result = reinterpret_cast<ULONG_PTR>(ShellExecuteW(
      nullptr, L"open", L"https://google.com" , nullptr, nullptr,
      SW_SHOWNORMAL));

  if (result == SE_ERR_NOASSOC)
    std::cout << "Error: No application is associated \n";

  std::cout << "Result : " << result;

  return 0;
}
```

**Comment by cawoodm** [comment]
I get no error when I run `require('electron').shell.openExternal('https://google.com')` - however Chrome just opens - there is no URL.

**Comment by cawoodm** [comment]
Found a solution here #84409.

**Comment by dbrennand** [comment]
Hi @deepak1556 

I will try this out ASAP.

Thanks 👍

**Comment by dbrennand** [comment]
Hi @deepak1556 
The following output was produced from the script:

```c++
Error: No application is associated 
Result : 31
```

**Comment by TimGangl** [comment]
was trying to create a path for MYSQL  for the command line. That is what broke it and I fixed it.

**Comment by dbrennand** [comment]
I am still having this issue. On version 1.42 now.

**Comment by boay24** [comment]
same here , open the external site not work.
OS Build 19569 insider.
![image](https://user-images.githubusercontent.com/61148495/75122963-dd5b6a80-56d5-11ea-9836-9980c4878894.png)

**Comment by octref** [comment]
@boay24 What does clicking "Open" do? Do you see any error in Help -> Developer Tools? What's the result of running: `require('electron').shell.openExternal('https://google.com')`?

**Comment by oneilonline** [comment]
VSC 1.43.0 Ubuntu 19.10 Same issue. Any extension with an external link to their website doesn't not open.

**Comment by dbrennand** [comment]
Indeed. This issue still persists.

```
Version: 1.43.0 (system setup)
Commit: 78a4c91400152c0f27ba4d363eb56d2835f9903a
Date: 2020-03-09T19:47:57.235Z
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363
```

**Comment by deepak1556** [comment]
@oneilonline this issue is wrt windows, its better to track your issue as a separate one. Before opening the issue please add the following details to the report.

* Can you run `xdg-open https://example.com` in your terminal and see if that brings up your default browser.

* If your default browser is chrome can you add the information based on https://github.com/microsoft/vscode/issues/92902#issuecomment-601008590


@boay24 what is the output of running the test executable from https://github.com/microsoft/vscode/issues/84265#issuecomment-554222723 ?


@dbrennand the output from https://github.com/microsoft/vscode/issues/84265#issuecomment-555977508 makes it obvious that an application for that mime type cannot be found on the system. There are two things we can do here,

1) Can you run the same executable as admin, just to verify if the default associated browser is just not available to the user you have first tested for.

2) A feature needs to be added in Electron that would perform the following equivalent

```
 auto result = reinterpret_cast<ULONG_PTR>(ShellExecuteW(
      nullptr, L"openas", L"https://google.com" , nullptr, nullptr,
      SW_SHOWNORMAL));
```

which will bring up `Open With` dialog to choose the application. Based on the output of 1) I will write down the next course of action here.

**Comment by dbrennand** [comment]
Hi @deepak1556 
The output I sent in [#84265 (comment)](https://github.com/microsoft/vscode/issues/84265#issuecomment-555977508) was from running as administrator.

**Comment by deepak1556** [comment]
Thanks for confirming @dbrennand , I am marking this issue as blocked on a feature update to the api from electron.

**Comment by bchuecos** [comment]
same issue here
```
Version: 1.49.3
Commit: 2af051012b66169dde0c4dfae3f5ef48f787ff69
Date: 2020-10-02T17:56:11.027Z
Electron: 9.2.1
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Linux x64 5.4.0-1026-azure
```
Default browser is set to firefox. Anytime I try to do anything that requires lauching a web browser from VS Code, nothing happens. This also affects extensions that require opening a browser. Running xdg-open https://example.com in the terminal opens the browser just fine.

**Comment by rzhao271** [comment]
@dbrennand Though I linked an upstream fix, I'm unable to reproduce the issue locally, so it can't be merged in yet.

I'm wondering whether this issue still occurs for you on the latest VS Code Insiders edition (v1.53.0-insider)? If it does, I'm wondering whether you can paste the full specs after running the `Help: About` command?

**Comment by dbrennand** [comment]
Hi @rzhao271 

It has been sometime since this issue and I no longer have the device which I was experiencing this issue on.

**Comment by rzhao271** [comment]
I'll close this issue for now.
If this issue shows up again, feel free to file a new issue.
